src/lib/features/instance-stats/instance-stats-service.ts

```diff
@@ -193,55 +196,95 @@ export class InstanceStatsService {
         this.trafficDataUsageStore = trafficDataUsageStore;
     }
 
+    memory = new Map<string, () => Promise<any>>();
+    memorize<T>(key: string, fn: () => Promise<T>): Promise<T> {
+        const enabled = this.flagResolver.isEnabled('memorizeStats', {
+            memoryKey: key,
+        });
+        if (enabled) {
+            const variant = this.flagResolver.getVariant('memorizeStats', {
```

Comment:
If we resolve the flag then check the variant after I think it does weird things to the metrics. Might be an excellent place to use `variant.feature_enabled` :sweat_smile:

Reply:
Oh! Nice suggestion! I wonder if this is clear from our docs :D

Reply:
Thanks! [33cbd0f](https://github.com/Unleash/unleash/pull/8766/commits/33cbd0f04bccb16b6d091869d1313c672f8304cd) should address that